tensorflow/python/kernel_tests/pooling_ops_test.py

```diff
@@ -1269,6 +1538,17 @@ def Test(self):
 
   return Test
 
+def GetMaxPoolGradGradTest(input_size, filter_size, output_size, strides, padding):
+
+  def Test(self):
+    # MaxPoolWithArgMax is implemented only on CUDA.
+    if not test.is_gpu_available(cuda_only=True):
+      return
+    self._CompareMaxPoolingGradBk(input_size, output_size, filter_size, strides,
+                                  padding)
+
+  return Test
+
 
 if __name__ == "__main__":
   for (name_, input_size_, filter_size_, output_size_, stride_,
```

Comment:
This is broken. The test names are not unique and therefore only one of the possible tests will get run. This should set a test name that is a concatenation of all the test params. I also recommend using a function like 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/matmul_op_test.py#L35

to add the tests so this will not happen in the future.